File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    static String binary(int n, int size){
        String ret = "";
        int temp = n;
        while(temp > 0){
            ret = "" + temp % 2 + ret;
            temp /= 2;
        }
        while(ret.length() < size){
            ret = "0" + ret;
        }
        return ret;
    }
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] ret = new String[n];
        String[] temp1 = new String[n];
        String[] temp2 = new String[n];
        for(int i = 0; i < n; i++){
            temp1[i] = binary(arr1[i], n);
            temp2[i] = binary(arr2[i], n);
            // System.out.println(temp1[i] + "::" + temp2[i]);
        }
        for(int i = 0; i < n; i++){
            String temp = "";
            for(int j = 0; j < n; j++){
                if(temp1[i].charAt(j) == '1' || temp2[i].charAt(j) == '1'){
                    temp += "#";
                } else temp += " ";
            }
            ret[i] = temp;
        }
        
        
        return ret;
    }
}

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def num_to_bin(n, size):
    ret = ''
    while n > 0:
        ret += str(n % 2)
        n //= 2
    return ("0"*size + ret[::-1])[-size:]

def solution(n, arr1, arr2):
    for i in range(len(arr1)):
        arr1[i] = num_to_bin(arr1[i], n)
        arr2[i] = num_to_bin(arr2[i], n)
    ret = []
    for i in range(n):
        temp = ''
        for x1, x2 in zip(arr1[i], arr2[i]):
            if x1 == '1' or x2 == '1':
                temp += '#'
            else: temp += ' '
        ret.append(temp)
    return ret

"""
1. 둘다 이진수로 변환
2. 둘중하나라고 1이면 #으로 해서 만들기


"""